master/internal/agent/agent.go

```diff
@@ -68,7 +96,31 @@ func (a *agent) Receive(ctx *actor.Context) error {
 		if err := ctx.Ask(a.socket, wsm).Error(); err != nil {
 			ctx.Log().WithError(err).Error("failed to write master set agent options")
 		}
+		if a.awaitingReconnect {
+			ctx.Log().Info("agent reconnected")
+			a.awaitingReconnect = false
+
+			// Re-propagate our old state back on successful recovery.
+			if a.enabled {
+				ctx.Tell(a.resourcePool, sproto.EnableAgent{Agent: ctx.Self()})
+			} else {
+				ctx.Tell(a.resourcePool, sproto.DisableAgent{Agent: ctx.Self(), Drain: a.draining})
+			}
+			ctx.Tell(a.slots, patchSlot{Enabled: a.enabled, Drain: a.draining})
+
+			for msg := range a.reconnectBacklog {
+				if err := a.receive(ctx, msg); err != nil {
+					return errors.Wrapf(err, "replaying backlog")
+				}
+			}
+			a.reconnectBacklog = nil
```

Comment:
is that supposed to reset it to empty array? I don't see where it's initialized

Reply:
since you can append to a nil array in golang, it doesn't need to be initialized. also, yeah, that's why i just set it back to nil.